### ISSUE TITLE
No good way to use leases with blobs

### ISSUE THREAD

**Issue by yaxia** [Azure/azure-storage-ruby#16]
Migrate from the Azure SDK: https://github.com/Azure/azure-sdk-for-ruby/issues/189

From @schadr: 
I am trying to use leases on blob to ensure that only one host can modify the content.

For that purpose I create a lease on a blob by using the 
 aquire_lease operation
 but there is no way to provide the lease_id for calls such as create_blob_pages, so the only way to use leases is to ensure a blob does not get modified while reading from it.

Or am I missing something?

**Comment by yaxia** [comment]
@schadr, we will take care of this in a later release. Thanks.

**Comment by pcibraro** [comment]
Any expected data for releasing this feature ? Thanks

**Comment by xiaomi7732** [comment]
Hi there,
Is there any progress on this issue? I would like to call create_block_blob with a lease id to update the leased block blob.

**Comment by yaxia** [comment]
It sounds another vote for this feature. I will prioritize this and have it in next release (expecting August, but not a commitment).

**Comment by sbonebrake** [comment]
+1 would also like this feature

**Comment by xiaomi7732** [comment]
@sbonebrake,
A workaround I am using here is to create a side file to lock upon. For example, if I intend to lock a.txt, I create a file 'a.txt.lock' and put a lease a.txt.lock. Then, we check the existence and the lease status of the lock file.
Just my 2 cents. Not saying that we should not fix this issue but might be a way to unblock a bit to make more progress for the project.

**Comment by sbonebrake** [comment]
Thanks, but this won't work for my project because I plan to have many different machines reading from the same container to process data from Event Hub capture. Fortunately I found another workaround:
```ruby
Azure::Storage::Service::StorageService.register_request_callback { |headers| headers["x-ms-lease-id"] = lease_id }
blob_client.delete_blob(container_name, blob.name)
Azure::Storage::Service::StorageService.register_request_callback { |headers| headers }
```

**Comment by yaxia** [comment]
We've added the support for per-request lease ID in [0.15.0](https://github.com/Azure/azure-storage-ruby/releases/tag/v0.15.0). Please check it out and any feedback is appreciated.
